v2/clients/interfaces/event.go

```diff
@@ -11,10 +11,33 @@ import (
 	"github.com/edgexfoundry/go-mod-core-contracts/errors"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/responses"
 )
 
 // EventClient defines the interface for interactions with the Event endpoint on the EdgeX Foundry core-data service.
 type EventClient interface {
-	// Add adds new events
-	Add(ctx context.Context, reqs []requests.AddEventRequest) ([]common.BaseWithIdResponse, errors.EdgeX)
+	// Add adds new event. The given deviceName and profileName must match to the deviceName and profileName of Event.
+	Add(ctx context.Context, profileName, deviceName string, req requests.AddEventRequest) (common.BaseWithIdResponse, errors.EdgeX)
```

Comment:
IMO, this wrapper should get `profileName` & `deviceName` from `req`, rather than make user supply it twice. Hides our implementation to uses them in the REST path.

Reply:
updated.